### PR TITLE
obs_decam/DM-7884: unification of coadd and diff datasets

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

File: policy/DecamMapper.paf [hunk 1]
```diff
@@ -96,37 +96,17 @@ exposures: {
         level:       "None"
         tables:      raw
     }
+    # dataset defined in obs_base modified in this mapper as follows:
```
Comment (TallJimbo) [inline policy/DecamMapper.paf]: I don't think there's any need to override `level` for this dataset.  I don't think it actually does anything in this context, and certainly if some other value of `level` works for other `obs_` packages that same value will work for this one.

File: policy/DecamMapper.paf [hunk 2]
```diff
@@ -421,57 +385,25 @@ datasets: {
...
-        storage:       "BoostStorage"
     }
+    # dataset defined in obs_base modified in this mapper as follows:
     deepCoadd_forced_src: {
         template:      "deepCoadd-results/%(filter)s/%(tract)d/%(patch)s/forcedSrc-%(filter)s-%(tract)d-%(patch)s.fits"
-        python:        "lsst.afw.table.SourceCatalog"
-        persistable:   "ignored"
-        storage:       "FitsCatalogStorage"
     }
+    # dataset defined in obs_base modified in this mapper as follows:
```
Comment (TallJimbo) [inline policy/DecamMapper.paf]: There's no reason for any one camera to be different from any other on `python` entries like this one, but I don't actually know whether this one is correct or incorrect.  Given that there has been more difference imaging work on obs_decam than other cameras recently, I suspect it's more likely this value is correct and obs_base is incorrect.  @SimonKrughoff, @ctslater, can you confirm or deny that this is the right config entry, or do you know who could?

File: policy/DecamMapper.paf (hunk 1 above)
Comment (pgee2000) [inline policy/DecamMapper.paf]: I will list this as a change under RFC-232

File: policy/DecamMapper.paf (hunk 2 above)
Comment (pgee2000) [inline policy/DecamMapper.paf]: I will change this in obs_base.  this should be lsst.pipe.tasks.imageDifference.ImageDifferenceConfig

**Pull request merged**